accounts/migrations/0002_auto_20200213_1711.py

```diff
@@ -0,0 +1,54 @@
+# Generated by Django 3.0.3 on 2020-02-13 22:11
+
+from django.db import migrations, transaction
+
+
+def create_groups(apps, schema_editor):
+    with transaction.atomic():
+        User = apps.get_model("accounts", "User")
+        Group = apps.get_model("auth", "Group")
+        for user in User.objects.all():
+            for affiliation in user.affiliation.all():
+                if affiliation.name:
+                    group, _ = Group.objects.get_or_create(name=affiliation.name)
+                    user.groups.add(group)
+            user.save()
+
+
+def copy_permissions(apps, schema_editor):
+    with transaction.atomic():
+        User = apps.get_model("accounts", "User")
+        Permission = apps.get_model("auth", "Permission")
+        for user in User.objects.all():
+            for product_permission in user.product_permission.all():
+                group = Permission.objects.get(codename=product_permission.id)
+                user.user_permissions.add(group)
+            user.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("accounts", "0001_initial"),
+    ]
+
+    operations = [
+        migrations.RunPython(create_groups),
+        migrations.RemoveField(model_name="user", name="affiliation",),
+        migrations.DeleteModel(name="PennAffiliation",),
+        migrations.AlterModelOptions(
+            name="user",
+            options={
+                "permissions": (
+                    ("cfa_admin", "Admin for Common Funding Application"),
+                    ("clubs_admin", "Admin for Penn Clubs"),
+                    ("courses_admin", "Admin for Penn Courses"),
+                    ("pcr_admin", "Admin for Penn Course Review"),
+                    ("studentlife_admin", "Admin for Student Life"),
```

Comment:
is there a way to add these dynamically or will it require a code change every time?

Reply:
We definitely could do it dynamically, but django seems to[ suggest doing it this way](https://docs.djangoproject.com/en/3.0/topics/auth/customizing/#custom-permissions). Honestly I don't know why or what the tradeoffs are though.